lib/active_merchant/billing/gateways/ipg.rb

```diff
@@ -160,7 +162,7 @@ def add_payment(xml, payment, options = {}, credit_envelope = 'v1')
           xml.tag!("#{credit_envelope}:CreditCardData") do
             xml.tag!('v1:CardNumber', payment.number) if payment.number
             xml.tag!('v1:ExpMonth', payment.month) if payment.month
-            xml.tag!('v1:ExpYear', payment.year) if payment.year
+            xml.tag!('v1:ExpYear', payment.year % 100) if payment.year && (payment.year.digits.count == 4)
```

Comment:
I'm not sure if this is what we want. It only adds year if it is passed in as 4 digits.

There is a method in `lib/active_merchant/billing/credit_card_formatting.rb` called `format` that can be called like this:

```
xml.tag!('v1:ExpYear', format(payment.year, :two_digits)) if payment.year
```

...that way we don't have to care about how many digits the year on the card is. This will send a 2-character string every time.

Reply:
ah yes. thats what I wanted to do. Thanks for informing, this change is updated in latest commit.